core/src/wallet/cosmos/explorers/RpcsParsers.hpp

```diff
@@ -508,6 +508,189 @@ namespace ledger {
+            //       "balance": "1850"
+            //     }
+            //   ]
+            // }
+            template <typename T>
+            void parseRedelegation(const T& n, cosmos::Redelegation &out) {
+                assert((n.HasMember(kDelegatorAddress)));
+                assert((n.HasMember(kValidatorSrcAddress)));
+                assert((n.HasMember(kValidatorDstAddress)));
+                assert((n.HasMember("entries")));
+                out.entries = std::list<cosmos::RedelegationEntry>();
+                out.delegatorAddress = n[kDelegatorAddress].GetString();
+                out.srcValidatorAddress = n[kValidatorSrcAddress].GetString();
+                out.dstValidatorAddress = n[kValidatorDstAddress].GetString();
+                for (const auto& entry : n["entries"].GetArray()) {
+                    cosmos::RedelegationEntry parsedEntry;
+                    parseRedelegationEntry(entry.GetObject(), parsedEntry);
+                    out.entries.emplace_back(std::move(parsedEntry));
+                }
+            }
+
+            // Parse redelegation (/staking/redelegations?delegator={address})
+            // into cosmos::RedelegationList
+            // {
+            //   "height": "1347368",
+            //   "result": [
+            //     {
+            //       "delegator_address": "cosmos1g84934jpu3v5de5yqukkkhxmcvsw3u2ajxvpdl",
+            //       "validator_src_address": "cosmosvaloper1sjllsnramtg3ewxqwwrwjxfgc4n4ef9u2lcnj0",
+            //       "validator_dst_address": "cosmosvaloper1clpqr4nrk4khgkxj78fcwwh6dl3uw4epsluffn",
+            //       "entries": [
+            //         {
+            //           "creation_height": 1107334,
+            //           "completion_time": "2020-04-01T15:46:03.941380099Z",
+            //           "initial_balance": "1850",
+            //           "shares_dst": "1850.000000000000000000",
+            //           "balance": "1850"
+            //         }
+            //       ]
+            //     },
+            //     {
+            //       "delegator_address": "cosmos1g84934jpu3v5de5yqukkkhxmcvsw3u2ajxvpdl",
+            //       "validator_src_address": "cosmosvaloper1ey69r37gfxvxg62sh4r0ktpuc46pzjrm873ae8",
+            //       "validator_dst_address": "cosmosvaloper1vf44d85es37hwl9f4h9gv0e064m0lla60j9luj",
+            //       "entries": [
+            //         {
+            //           "creation_height": 1178354,
+            //           "completion_time": "2020-04-07T13:08:44.940330001Z",
+            //           "initial_balance": "1000000",
+            //           "shares_dst": "1000000.000000000000000000",
+            //           "balance": "1000000"
+            //         }
+            //       ]
+            //     }
+            //   ]
+            // }
+            template <typename T>
+            void parseRedelegationList(const T& n, cosmos::RedelegationList &out) {
```

Comment:
Can't this be used in `GaiaCosmosLikeBlockchainExplorer::getRedelegationsByDelegator`?

Reply:
Yup, no idea why I didn't do it